flocker/testtools/__init__.py

```diff
@@ -299,6 +302,28 @@ def loop(failure):
     return d
 
 
+def retry_effect_with_timeout(effect, timeout):
+    """
+    If ``effect`` fails, retry it until ``timeout`` expires.
+
+    :param Effect effect: The Effect to retry.
+    :param int timeout: Timeout in seconds.
+    """
+
+    end_time = time.time() + timeout
+
+    def ret(e):
```

Comment:
`ret` here stands for "retry"? Please rename it to `retry` to avoid confusion (`ret` is often used to mean "return value").